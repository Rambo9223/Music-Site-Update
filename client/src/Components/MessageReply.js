import { Reply } from "react-bootstrap-icons";
/* The link to set up the email for a reply from message centre */
export default function MessageReply (props){
    let message = props.message
    return( 
    <>
    <span className="link"><a href={`mailto:${message.email}?subject=${message.subject}&body=${encodeURIComponent(message.enquiry)}`}><Reply/> Reply</a></span>
    </>
    )
}
