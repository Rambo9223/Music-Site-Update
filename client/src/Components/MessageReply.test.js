/**
*@jest-environment jsdom
*/
import { render,cleanup} from '@testing-library/react';
import * as React from 'react';
import MessageReply from './MessageReply';

afterEach(()=>{
    cleanup();
});

let message = {
    email:"testemail@mail.com",
    subject:"Can You Read This",
    enquiry:"This is a test email props for the message reply test."
}

describe('Message Reply Tests', () => {

    test('Page matches snapshot and element exists with props passed', async () => {
        let page = render(<MessageReply message={message}/>)
        expect(page).toMatchSnapshot();
        let link = await page.findByText("Reply",{exact:false});
        expect(link.outerHTML.includes(message.email)).toEqual(true);
        expect(link).toBeInTheDocument();
    })
})