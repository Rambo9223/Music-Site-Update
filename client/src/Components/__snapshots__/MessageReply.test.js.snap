// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Reply Tests Page matches snapshot and element exists with props passed 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="link"
      >
        <a
          href="mailto:testemail@mail.com?subject=Can You Read This&body=This%20is%20a%20test%20email%20props%20for%20the%20message%20reply%20test."
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.598 5.013a.144.144 0 0 1 .202.134V6.3a.5.5 0 0 0 .5.5c.667 0 2.013.005 3.3.822.984.624 1.99 1.76 2.595 3.876-1.02-.983-2.185-1.516-3.205-1.799a8.7 8.7 0 0 0-1.921-.306 7 7 0 0 0-.798.008h-.013l-.005.001h-.001L7.3 9.9l-.05-.498a.5.5 0 0 0-.45.498v1.153c0 .108-.11.176-.202.134L2.614 8.254l-.042-.028a.147.147 0 0 1 0-.252l.042-.028zM7.8 10.386q.103 0 .223.006c.434.02 1.034.086 1.7.271 1.326.368 2.896 1.202 3.94 3.08a.5.5 0 0 0 .933-.305c-.464-3.71-1.886-5.662-3.46-6.66-1.245-.79-2.527-.942-3.336-.971v-.66a1.144 1.144 0 0 0-1.767-.96l-3.994 2.94a1.147 1.147 0 0 0 0 1.946l3.994 2.94a1.144 1.144 0 0 0 1.767-.96z"
            />
          </svg>
           Reply
        </a>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="link"
    >
      <a
        href="mailto:testemail@mail.com?subject=Can You Read This&body=This%20is%20a%20test%20email%20props%20for%20the%20message%20reply%20test."
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.598 5.013a.144.144 0 0 1 .202.134V6.3a.5.5 0 0 0 .5.5c.667 0 2.013.005 3.3.822.984.624 1.99 1.76 2.595 3.876-1.02-.983-2.185-1.516-3.205-1.799a8.7 8.7 0 0 0-1.921-.306 7 7 0 0 0-.798.008h-.013l-.005.001h-.001L7.3 9.9l-.05-.498a.5.5 0 0 0-.45.498v1.153c0 .108-.11.176-.202.134L2.614 8.254l-.042-.028a.147.147 0 0 1 0-.252l.042-.028zM7.8 10.386q.103 0 .223.006c.434.02 1.034.086 1.7.271 1.326.368 2.896 1.202 3.94 3.08a.5.5 0 0 0 .933-.305c-.464-3.71-1.886-5.662-3.46-6.66-1.245-.79-2.527-.942-3.336-.971v-.66a1.144 1.144 0 0 0-1.767-.96l-3.994 2.94a1.147 1.147 0 0 0 0 1.946l3.994 2.94a1.144 1.144 0 0 0 1.767-.96z"
          />
        </svg>
         Reply
      </a>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
